assembl/static2/js/app/utils/urlPreview.jsx

```diff
@@ -0,0 +1,205 @@
+/* eslint-disable no-param-reassign, guard-for-in, no-restricted-syntax */
+// @flow
+import React from 'react';
+import type { Node } from 'react';
+
+import YoutubeEmbed from '../components/common/urlPreview/youtubeTheater';
+import SketchFabEmbed from '../components/common/urlPreview/sketchFabTheater';
+
+const urlMetadataWebServiceUrl = 'http://0.0.0.0:5000';
+
+type MetadataResponsProps = {
+  code: string,
+  metadata: Object
+};
+
+type URLMetadataProps = {
+  providerName: string,
+  html: string,
+  id: string
+};
+
+export type Script = {
+  src: string,
+  type: 'url' | 'js'
+};
+
+export const EMBED_PROVIDERS = {
+  youtube: {
+    scheme: /(?:youtube\.com\/(?:[^/]+\/.+\/|(?:v|e(?:mbed)?)\/|.*[?&]v=)|youtu\.be\/)([^"&?/ ]{11})/i,
+    match: (url: string) => url.match(EMBED_PROVIDERS.youtube.scheme),
+    getMetadata: (url: string): URLMetadataProps | null => {
+      const result = EMBED_PROVIDERS.youtube.match(url);
+      if (result) {
+        const id = result[1];
+        const embedUrl = `https://www.youtube.com/embed/${id}`;
+        return {
+          providerName: 'Youtube',
+          html: `<div><iframe title="" src="${embedUrl}" frameborder="0" class="embedded-video"
+                              allowfullscreen data-source-url="${url}"></iframe></div>`,
+          id: id
+        };
+      }
+      return null;
+    },
+    getComponent: (url: string): Node | null => {
+      const result = EMBED_PROVIDERS.youtube.match(url);
+      return result ? <YoutubeEmbed id={result[1]} /> : null;
+    }
+  },
+  sketchfab: {
+    scheme: /(?:((sketchfab\.com\/show|models)|skfb\.ly)\/(.*))/i,
+    match: (url: string) => url.match(EMBED_PROVIDERS.sketchfab.scheme),
+    getMetadata: (url: string): URLMetadataProps | null => {
+      const result = EMBED_PROVIDERS.sketchfab.match(url);
+      if (result) {
+        const isSkfb = result[1] === 'skfb.ly';
+        const id = result[3];
+        const embedUrl = isSkfb
+          ? `${url}?autostart=1&autospin=0.5`
+          : `https://sketchfab.com/models/${id}/embed?autostart=1&autospin=0.5`;
+
+        return {
+          providerName: 'Sketchfab',
+          html: `<div><iframe title="" src="${embedUrl}" frameborder="0" class="embedded-video"
+                              allowfullscreen data-source-url="${url}"></iframe></div>`,
+          id: id
+        };
+      }
+      return null;
+    },
+    getComponent: (url: string): Node | null => {
+      const result = EMBED_PROVIDERS.sketchfab.match(url);
+      return result ? <SketchFabEmbed url={url} /> : null;
+    }
+  }
+  // add ...
+};
+
+/**
+ * @param {dict: Object | null} The dict to format
+ * @returns {Object} Returns a copy of the dict with formatted keys (ex: {first_name: value} -> {firstName: value}).
+ */
+function formatDict(dict: Object | null): Object {
```

Comment:
In js, we call this an object, not a dict. `transformKeysToCamelCase` may be a better name?